CIP-????/README.md

```diff
@@ -0,0 +1,90 @@
+---
+CIP: ???
+Title: multi-arity uplc lambda node
+Category: Plutus
+Status: Proposed
+Authors:
+    - Michele Nuzzi <michele@harmoniclabs.tech>
+Implementors: []
+Discussions:
+    - https://github.com/cardano-foundation/CIPs/pull/1043
+Created: 2025-06-02
+License: CC-BY-4.0
+---
+
+<!-- Existing categories:
+
+- Meta     | For meta-CIPs which typically serves another category or group of categories.
+- Wallets  | For standardisation across wallets (hardware, full-node or light).
+- Tokens   | About tokens (fungible or non-fungible) and minting policies in general.
+- Metadata | For proposals around metadata (on-chain or off-chain).
+- Tools    | A broad category for ecosystem tools not falling into any other category.
+- Plutus   | Changes or additions to Plutus
+- Ledger   | For proposals regarding the Cardano ledger (including Reward Sharing Schemes)
+- Catalyst | For proposals affecting Project Catalyst / the Jörmungandr project
+
+-->
+
+## Abstract
+<!-- A short (\~200 word) description of the proposed solution and the technical issue being addressed. -->
+This CIP proposes a new node for an uplc tree.
+
+The node is intended to have the same effect of n consecutive lambda nodes for 1/n the cost. Improving contracts in all measurable metrics (size, cpu, memory)
+
+## Motivation: why is this CIP necessary?
+<!-- A clear explanation that introduces the reason for a proposal, its use cases and stakeholders. If the CIP changes an established design then it must outline design issues that motivate a rework. For complex proposals, authors must write a Cardano Problem Statement (CPS) as defined in CIP-9999 and link to it as the `Motivation`. -->
+
+Patterns like
+```uplc
+(lam a (lam b (lam c <body>)))
+```
+are increasingly common in uplc contracts.
+
+These are not only used to represent functions with arity greater than 1, but also to destructure SoP enocded values, or to introduce multiple "letted" values in scope (witch in plutus v3 is done using a case/constr).
```

Comment:
after reading a few facts:
- such patterns are common;
- they are used for X, Y, Z...

I struggle to see how this answers the question "Why is this CIP necessary"?
What is the problem being fixed?